src/baobab.d.ts

```diff
@@ -0,0 +1,205 @@
+import Emitter from "emmett";
+
+interface PlainObject<T = any> {
+  [key: string]: T;
+}
+
+type Predicate = (data: any) => boolean;
+type Constraints = object;
```

Comment:
Shouldn't this be a `PlainObject`?

Reply:
Yes indeed.

Reply:
I just noticed that there are actually a lot of remaining `object` that should be removed.

Reply:
Done.